kernel/byterun/coq_interp.c

```diff
@@ -76,6 +76,14 @@ sp is a local copy of the global variable extern_sp. */
 #   define print_lint(i)
 #endif
 
+#define CHECK_STACK(num_args) {                                                \
+if (sp - num_args < coq_stack_threshold) {                                     \
+  coq_sp = sp;                                                                 \
+  realloc_coq_stack(num_args * sizeof(value) + Coq_stack_threshold);           \
```

Comment:
This seems dubious. `realloc_coq_stack` takes as argument the number of words to allocate. So `num_args` should not be multiplied by `sizeof(value)`.